File: src/foam/core/Listener.java
/**
 * @license
 * Copyright 2017 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

package foam.core;

public interface Listener {
  void apply(foam.core.Detachable sub, java.util.List args);
}

File: src/foam/cross_platform/Listener.js
foam.INTERFACE({
  package: 'foam.cross_platform',
  name: 'Listener',
  methods: [
    {
      name: 'executeListener',
      args: [
        {
          type: 'foam.core.Detachable',
          name: 'sub'
        },
        {
          type: 'Any[]',
          name: 'args'
        }
      ]
    }
  ]
});
